File: spring-webflux/src/main/java/software/darkmatter/school/blog/domain/comment/data/CommentRepository.java
package software.darkmatter.school.blog.domain.comment.data;

import org.springframework.data.domain.Pageable;
import org.springframework.data.repository.reactive.ReactiveCrudRepository;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface CommentRepository extends ReactiveCrudRepository<Comment, Long> {

    Mono<Comment> findByIdAndDeletedAtIsNull(Long id);

    Flux<Comment> findAllByPostIdAndDeletedAtIsNull(Long postId, Pageable pageable);
}

File: kotlin-spring-webflux/src/main/kotlin/software/darkmatter/school/blog/domain/comment/data/CommentRepository.kt
package software.darkmatter.school.blog.domain.comment.data

import kotlinx.coroutines.flow.Flow
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.reactive.ReactiveCrudRepository

interface CommentRepository : ReactiveCrudRepository<Comment, Long> {

    suspend fun findByIdAndDeletedAtIsNull(id: Long?): Comment?

    fun findAllByPostIdAndDeletedAtIsNull(postId: Long, pageable: Pageable): Flow<Comment>

    suspend fun findByPostIdAndIdAndDeletedAtIsNull(postId: Long, id: Long): Comment?
}
